code/go/0chain.net/core/common/utils.go

```diff
@@ -22,3 +25,15 @@ func ToKey(key interface{}) string {
 func IsEqual(key1, key2 string) bool {
 	return key1 == key2
 }
+
+// getParentPaths For path /a/b/c.txt, will return [/a,/a/b]
+func GetParentPaths(fPath string) []string {
+	splittedPaths := strings.Split(fPath, "/")
+	var paths []string
+
+	for i := 0; i < len(splittedPaths); i++ {
+		subPath := strings.Join(splittedPaths[0:i], "/")
+		paths = append(paths, subPath)
+	}
+	return paths[2:]
```

Comment:
This will panic if fPath is "". Please add unit tests to check the edge cases. 

Also, do you think we should consider the cases such as "/a//b", and //a//b, etc?

Reply:
I've added `filepath.Clean()` in this function and removed it from `validateParentType` function.

For above cases; I think its necessary to cleaning path before processing. It should be done in functions as early as relevant and we should check against if filepath is not absolute.